delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/OrganizationsDummy.scala

```diff
@@ -0,0 +1,172 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.testkit
+
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicLong
+
+import akka.persistence.query.{NoOffset, Offset}
+import cats.effect.Clock
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, Label}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.organizations.OrganizationCommand.{CreateOrganization, DeprecateOrganization, UpdateOrganization}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.organizations.OrganizationRejection.{RevisionNotFound, UnexpectedInitialState}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.organizations.OrganizationState.Initial
+import ch.epfl.bluebrain.nexus.delta.sdk.model.organizations.{OrganizationCommand, OrganizationRejection, OrganizationState, _}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.ResultEntry.UnscoredResultEntry
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchParams.OrganizationSearchParams
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.UnscoredSearchResults
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.{Pagination, SearchResults}
+import ch.epfl.bluebrain.nexus.delta.sdk.testkit.OrganizationsDummy._
+import ch.epfl.bluebrain.nexus.delta.sdk.{OrganizationResource, Organizations}
+import ch.epfl.bluebrain.nexus.testkit.{IORef, IOSemaphore}
+import fs2.Stream
+import monix.bio.{IO, Task, UIO}
+
+/**
+  * A dummy Organizations implementation that uses a synchronized in memory journal.
+  *
+  * @param journal     a ref to the journal containing all the events wrapped in an envelope
+  * @param cache       a ref to the cache containing all the current organization resources
+  * @param uuidToLabel a ref to the map where the keys are the uuids and the values are the labels
+  * @param semaphore   a semaphore for serializing write operations on the journal
+  * @param maxStreamSize    truncate event stream after this size
+  */
+final class OrganizationsDummy private (
+    journal: IORef[OrganizationsJournal],
+    cache: IORef[OrganizationsCache],
+    uuidToLabel: IORef[UUIDToLabel],
+    semaphore: IOSemaphore,
+    maxStreamSize: Long
+)(implicit clock: Clock[UIO])
+    extends Organizations {
+
+  private val offsetMax = new AtomicLong()
+
+  private def journalMap: UIO[Map[Label, OrganizationsJournal]] =
+    journal.get.map { v =>
+      v.groupBy(_.event.label).map {
+        case (k, v) =>
+          k -> v.sortBy(_.sequenceNr)
+      }
+    }
+
+  override def create(
+      label: Label,
+      description: Option[String]
+  )(implicit caller: Subject): IO[OrganizationRejection, OrganizationResource] =
+    eval(CreateOrganization(label, UUID.randomUUID(), description, caller)).flatMap(setUuid)
```

Comment:
In projects as generation uuid is a side effect just as getting current date, I removed the uuid from the create command and added an implicit `UIO[UUID]` besides the Clock